plugins/modules/dcnm_vrf.py

```diff
@@ -342,15 +404,40 @@ def diff_for_attach_deploy(self, want_a, have_a):
             if have_a:
                 for have in have_a:
                     if want['serialNumber'] == have['serialNumber']:
-                        found = True
+                        if want['extensionValues'] != "" and have['extensionValues'] != "":
+                            want_ext_values = want['extensionValues']
+                            want_ext_values = eval(want_ext_values)
```

Comment:
Why are you using `eval` here?  There are some inherent security risks for eval so I think we should look for a different approach.

See: 
* https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html
* https://realpython.com/python-eval-function/#minimizing-the-security-issues-of-eval

Reply:
I am using it for converting pupose. Will need to look for better approach, may be we can discuss.

Reply:
This is still something we need to change.  We should not use eval.

Possible alternative:  https://eulertech.wordpress.com/2018/06/10/the-pitfall-of-eval-function-and-its-safe-alternative-in-python/

Reply:
done, changed it with ast.literal_eval.

Reply:
Tested the code with UT and IT after the eval change to ast.literal_eval, all success.